src/main/java/com/umc/networkingService/domain/board/service/BoardServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    @Override
+    public BoardDetailResponse showBoardDetail(Member member, UUID boardId) {
+
+        Board board = loadEntity(boardId);
+
+        //조회수 증가
+        board.increaseHitCount();
+
+        //해당 게시글의 모든 첨부파일 url
+        List<String> boardFiles = boardFileService.findBoardFiles(board).stream()
+                .map(BoardFile::getUrl).toList();
+
+        return boardMapper.toBoardDetailResponse(board, boardFiles);
+
+    }
+
+    @Override
+    public BoardPagingResponse searchBoard(Member member, String keyword, Pageable pageable) {
+
+        return boardMapper.toBoardPagingResponse(boardRepository.findKeywordBoards(member, keyword, pageable));
+
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse toggleBoardLike(Member member, UUID boardId) {
+        Board board = loadEntity(boardId);
+
+        //boardHeart에 없다면 새로 저장
+        BoardHeart boardHeart = boardHeartRepository.findByMemberAndBoard(member, board)
+                .orElseGet(() -> {
+                    BoardHeart newHeart = boardHeartMapper.toBoardHeartEntity(board, member);
+                    boardHeartRepository.save(newHeart);
+                    board.addBoardHeart(newHeart);
+                    return newHeart;
+                });
+
+        boardHeart.toggleHeart();
+        board.setHeartCount(boardHeart.isChecked());
+
+        return new BoardIdResponse(boardId);
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse createBoard(Member member, BoardCreateRequest request, List<MultipartFile> files) {
+        //연합, 지부, 학교 타입
+        HostType hostType = request.getHostType();
+        BoardType boardType = request.getBoardType();
+
+        //boardType과 HostType에 따라 권한 판단
+        List<Semester> semesterPermission = checkPermission(member, hostType, boardType);
+
+        Board board = boardRepository.save(boardMapper.toEntity(member,
+                request.getTitle(),
+                request.getContent(),
+                request.getHostType(),
+                request.getBoardType(),
+                semesterPermission));
+
+        if (files != null)
+            boardFileService.uploadBoardFiles(board, files);
+
+        return new BoardIdResponse(board.getId());
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse updateBoard(Member member, UUID boardId, BoardUpdateRequest request, List<MultipartFile> files) {
+
+        Board board = loadEntity(boardId);
+
+        //연합, 지부, 학교 타입
+        HostType hostType = request.getHostType();
+        BoardType boardType = request.getBoardType();
+
+        //boardType과 HostType에 따라 권한 판단
+        List<Semester> semesterPermission = checkPermission(member, hostType, boardType);
+
+        board.update(request.getHostType(),
+                request.getBoardType(),
+                request.getTitle(),
+                request.getContent(),
+                semesterPermission);
```

Comment:
파라미터에 dto 단위로 받는 것이 가독성과 유지 보수에 좋을 거 같아요!!!
```suggestion
        board.update(request, semesterPermission);

```